### PR TITLE
Include product variations

### PR DESCRIPTION
**Pull request by macbookandrew**
adds support for creating dependencies on specific variations instead of just main products

### CODE REVIEW

**Comment by franticpsyx** [comment]
Hey @macbookandrew - thanks for the PR, this is a very welcome addition. Have been thinking a while now about adding this along with the ability to create dependencies at variation level, but haven't found the time just yet.

I will take a look as soon as possible and send some feedback or merge 👍

File: woocommerce-product-dependencies.php [hunk 1]
```diff
@@ -278,23 +279,43 @@ public function woo_tied_evaluate_access( $item_id ) {
...
+                            } elseif ( 'product_variation' == get_post_type( $id ) ) {
+                                $post_custom_metadata = get_post_custom( $id );
+                                // filter out everything except attributes
+                                $attributes = array();
+                                foreach ( $post_custom_metadata as $key => $value ) {
+                                    if ( strpos( $key, 'attribute_' ) === 0 ) {
+                                        $attributes['key'] = implode( ', ', $value );
+                                    }
+                                }
+                                $variations_name = implode( ', ', $attributes );
```
Comment (franticpsyx) [inline woocommerce-product-dependencies.php]: @macbookandrew For the sake of simplicity and writing easier to maintain code, I'd rather use a WC core method/function here, such as `WC_Product_Variation::get_formatted_variation_attributes`.

File: woocommerce-product-dependencies.php [hunk 2]
```diff
@@ -278,23 +279,43 @@ public function woo_tied_evaluate_access( $item_id ) {
...
+
+                                if ( $tied_product_ids[ 0 ] === $id ) {
+                                    $product_titles[] = sprintf( __( '&ldquo;<a href="%2$s">%1$s</a>&rdquo;', 'woocommerce-product-dependencies' ), $tied_products[ $id ]->get_title() . '&mdash;' . $variations_name . ' variation', get_permalink( $tied_products[ $id ]->id ) );
```
Comment (franticpsyx) [inline woocommerce-product-dependencies.php]: @macbookandrew Here, I would suggest integrating the extra wording inside the translatable strings. Raw string parts such as `&mdash;` or `variation` will remain untranslated.

File: woocommerce-product-dependencies.php [hunk 3]
```diff
@@ -112,11 +112,12 @@ public function woo_tied_products_write_panel() {
 				<p class="form-field">
 					<label>
 						<?php _e( 'Product Dependencies', 'woocommerce-product-dependencies' ); ?>
+						<?php _e( '(Use ID numbers to search for variations)', 'woocommerce-product-dependencies' ); ?>
```
Comment (franticpsyx) [inline woocommerce-product-dependencies.php]: This is a bit too verbose for my liking - perhaps you could use something like this to add a tooltip that is backwards compatible with WC versions before 2.5:

```
	/**
	 * Display a WooCommerce help tip.
	 *
	 * @param  string  $tip
	 * @return string
	 */
	public static function wc_help_tip( $tip ) {

		if ( version_compare( WC()->version, '2.5', '>=' ) ) {
			return wc_help_tip( $tip );
		} else {
			return '<img class="help_tip woocommerce-help-tip" data-tip="' . $tip . '" src="' . WC()->plugin_url() . '/assets/images/help.png" />';
		}
	}
```

File: woocommerce-product-dependencies.php (hunk 1 above)
Comment (macbookandrew) [inline woocommerce-product-dependencies.php]: @franticpsyx Should be better now—thanks for the suggestion.

File: woocommerce-product-dependencies.php (hunk 2 above)
Comment (macbookandrew) [inline woocommerce-product-dependencies.php]: @franticpsyx Thanks—this is my first time really working with translations.

File: woocommerce-product-dependencies.php (hunk 3 above)
Comment (macbookandrew) [inline woocommerce-product-dependencies.php]: @franticpsyx How about the update I just pushed?

**Pull request closed without merging**